File: Sort Colors/Sort_Colors.java
class Solution {
    public void sortColors(int[] nums) {
        int first = 0,second = nums.length-1;
        int i = 0;
        while(i<=second) {
            if(nums[i] == 0) {
                int temp = nums[i];
                nums[i] = nums[first];
                nums[first] = temp;
                first++;
                i++;
            }
            else if(nums[i] == 2) {
                int temp = nums[second];
                nums[second] = nums[i];
                nums[i] = temp;
                second--;
            }
            else {
                i++;
            }
        }
        return;
    }
}

File: Sort Colors/Sort_Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        first, second = 0, len(nums) - 1
        i = 0
        while i <= second:
            if nums[i] == 0:
                nums[i], nums[first] = nums[first], nums[i]
                first += 1
                i += 1
            elif nums[i] == 2:
                nums[i], nums[second] = nums[second], nums[i]
                second -= 1
            else:
                i += 1
        """
        Do not return anything, modify nums in-place instead.
        """
        